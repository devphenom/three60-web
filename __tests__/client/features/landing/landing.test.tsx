import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Landing from '@features/landing/landing';

describe('Landing', () => {
  it('should render landing without crashing', () => {
    render(<Landing />);
    const heading = screen.getByText(`Track how far you've gone`);
    const lists = screen.getAllByTestId('landing-lists');
    const description = screen.getByTestId('description');

    expect(heading).toBeInTheDocument();
    expect(lists).toHaveLength(4);
    expect(description).toBeInTheDocument();

    expect(heading).toMatchSnapshot();
    expect(lists).toMatchSnapshot();
    expect(description.textContent).toMatchSnapshot();
  });
});
