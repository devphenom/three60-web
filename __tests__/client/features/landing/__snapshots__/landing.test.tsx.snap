// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing should render landing without crashing 1`] = `
<h1
  class="chakra-heading css-0"
>
  Track how far 
  <br />
   you've gone
</h1>
`;

exports[`Landing should render landing without crashing 2`] = `
[
  <span
    class="css-0"
    data-testid="landing-lists"
  >
    <span
      style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
    >
      <span
        style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
      >
        <img
          alt=""
          aria-hidden="true"
          src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2720%27%20height=%2720%27/%3e"
          style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
        />
      </span>
      <img
        alt="todos"
        data-nimg="intrinsic"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
      />
      <noscript />
    </span>
  </span>,
  <span
    class="css-0"
    data-testid="landing-lists"
  >
    <span
      style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
    >
      <span
        style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
      >
        <img
          alt=""
          aria-hidden="true"
          src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2720%27%20height=%2720%27/%3e"
          style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
        />
      </span>
      <img
        alt="todos"
        data-nimg="intrinsic"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
      />
      <noscript />
    </span>
  </span>,
  <span
    class="css-0"
    data-testid="landing-lists"
  >
    <span
      style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
    >
      <span
        style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
      >
        <img
          alt=""
          aria-hidden="true"
          src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2720%27%20height=%2720%27/%3e"
          style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
        />
      </span>
      <img
        alt="todos"
        data-nimg="intrinsic"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
      />
      <noscript />
    </span>
  </span>,
  <span
    class="css-0"
    data-testid="landing-lists"
  >
    <span
      style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
    >
      <span
        style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
      >
        <img
          alt=""
          aria-hidden="true"
          src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2720%27%20height=%2720%27/%3e"
          style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
        />
      </span>
      <img
        alt="todos"
        data-nimg="intrinsic"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
      />
      <noscript />
    </span>
  </span>,
]
`;

exports[`Landing should render landing without crashing 3`] = `"Sit nisi incididunt tempor do duis fugiat proident excepteur. Ex elit pariatur incididunt nostrud occaecat mollit id occaecat. Voluptate adipisicing commodo."`;
